### DISCUSSION TITLE
How can I get the language of a Font in Avalonia?

### DISCUSSION THREAD

**Discussion by kyuranger** [AvaloniaUI/Avalonia#11348] [Q&A] [answered]
In WPF I can use the 
`FontFamily.FamilyNames.Keys`
to achieve this.

It seems Avalonia has no Keys property in FamilyNames.

What I want to do is to group font by its language. Word/photoshop and any other commercial software all achieve this, just like below for example of photoshop.

![image](https://github.com/AvaloniaUI/Avalonia/assets/30720036/e9711d48-24e0-4c26-b72d-56201cd437a4)

It is useful for non-English speaking countries to filter fonts for their language. How can I achieve this?

**Comment by yangjieshao** [comment]
https://github.com/AvaloniaUI/Avalonia/issues/11100

类似的问题 没得回我
(Similar issues)

**Reply by kyuranger** [reply]
> #11100
> 
> 类似的问题 没得回我 (Similar issues)

之前在Preview6的时候，Avalonia获取系统全部字体时候，声明中文字体所使用的是它的英文名，这样会导致字体无法被正确声明从而无法使用。后来我反馈过这个问题。Preview7不知道有没有被修复，但是我发现Preview8里面这个问题被修复了。
(In Preview6, when Avalonia obtained all fonts in the system, it declared that the Chinese font used its English name, which would cause the font to not be declared correctly and thus unusable. Later I reported this problem. I don't know if Preview7 has been fixed, but I found that this problem has been fixed in Preview8.)

**Comment by Gillibald** [comment, marked as answer]
A font has no language it just has family names mapped to locales so you can display the family name in the user's language. We currently don't have access to that name table but this will be fixed in the short future.

**Reply by kyuranger** [reply]
> A font has no language it just has family names mapped to locales so you can display the family name in the user's language. We currently don't have access to that name table but this will be fixed in the short future.

All right, thank you.
